packages/core-flows/src/price-list/workflows/batch-price-list-prices.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  BatchPriceListPricesWorkflowDTO,
+  BatchPriceListPricesWorkflowResult,
+} from "@medusajs/types"
+import {
+  WorkflowData,
+  createWorkflow,
+  parallelize,
+  transform,
+} from "@medusajs/workflows-sdk"
+import {
+  createPriceListPricesStep,
+  removePriceListPricesStep,
+  updatePriceListPricesStep,
+  validatePriceListsStep,
+  validateVariantPriceLinksStep,
+} from "../steps"
+
+export const batchPriceListPricesWorkflowId = "batch-price-list-prices"
+export const batchPriceListPricesWorkflow = createWorkflow(
+  batchPriceListPricesWorkflowId,
+  (
+    input: WorkflowData<{
+      data: BatchPriceListPricesWorkflowDTO
+    }>
+  ): WorkflowData<BatchPriceListPricesWorkflowResult> => {
+    const [_, variantPriceCreateMap, variantPriceUpdateMap] = parallelize(
+      validatePriceListsStep([{ id: input.data.id }]),
+      validateVariantPriceLinksStep([{ prices: input.data.create }]),
+      validateVariantPriceLinksStep([{ prices: input.data.update }]).config({
+        name: "variant-price-link-map-update",
+      })
+    )
+
+    const createPriceListPricesInput = transform(
+      { data: input.data, map: variantPriceCreateMap },
+      ({ data, map }) => ({
+        data: [{ id: data.id, prices: data.create }],
```

Comment:
Maybe this normalization can be done at the HTTP layer instead, and we always expect that the price set ID is passed in the workflow?

Reply:
The id here is the price list id, I think its ok to transform a bit here to make it work for other workflows.

Reply:
@riqwan yeah either way works, I was just thinking what we'd expect from the input, if you prefer it here then that's fine with me 👍